java/com/google/flatbuffers/Table.java

```diff
@@ -273,6 +284,320 @@ protected static int compareStrings(int offset_1, byte[] key, ByteBuffer bb) {
     }
     return len_1 - len_2;
   }
+
+  protected int __lookupByBooleanKey(int vectorFieldOffset, int keyFieldOffset, boolean queryKey,
+          boolean defaultValue ) {
+    byte key = (byte)(queryKey ? 1 : 0);
+    byte byteDefaultValue = (byte)(defaultValue ? 1 : 0);
+    return __lookupByByteKey(vectorFieldOffset, keyFieldOffset, key, byteDefaultValue);
+  }
+
+  protected int __lookupByByteKey(int vectorFieldOffset, int keyFieldOffset, int queryKey,
```

Comment:
The amount of changes appear pretty minimal, most of the code is identical. Besides maintenance, the code bloat is worse, since now _every_ user of FlatBuffers gets 11 copies of this function, versus only the users of the `key` feature. That is likely a lot less on average. Also, if we want to update C# to match Java, we need yet another 11 copies of this code, whereas with the generated code this is much easier to keep in sync.